src/main/java/com/arpnetworking/metrics/mad/Aggregator.java

```diff
@@ -271,124 +341,121 @@ public Key getKey() {
+                    .matchEquals(SHUTDOWN_MESSAGE, ignored -> this.shutdown())
                     .build();
         }
 
-        private List<ActorRef> createActors(final Key key) {
-            final List<ActorRef> periodWorkerList = Lists.newArrayListWithExpectedSize(_aggregator._periods.size());
-            for (final Duration period : _aggregator._periods) {
-                final Bucket.Builder bucketBuilder = new Bucket.Builder()
-                        .setKey(key)
-                        .setSpecifiedCounterStatistics(_aggregator._specifiedCounterStatistics)
-                        .setSpecifiedGaugeStatistics(_aggregator._specifiedGaugeStatistics)
-                        .setSpecifiedTimerStatistics(_aggregator._specifiedTimerStatistics)
-                        .setDependentCounterStatistics(_aggregator._dependentCounterStatistics)
-                        .setDependentGaugeStatistics(_aggregator._dependentGaugeStatistics)
-                        .setDependentTimerStatistics(_aggregator._dependentTimerStatistics)
-                        .setSpecifiedStatistics(_aggregator._cachedSpecifiedStatistics)
-                        .setDependentStatistics(_aggregator._cachedDependentStatistics)
-                        .setPeriod(period)
-                        .setSink(_aggregator._sink);
-                periodWorkerList.add(
-                        _aggregator._actorSystem.actorOf(
-                                Props.create(
-                                        PeriodWorker.class,
-                                        this.self(),
-                                        key,
-                                        period,
-                                        _aggregator._idleTimeout,
-                                        bucketBuilder,
-                                        _aggregator._periodicMetrics)));
+        private void shutdown() {
+            try {
+                // Start period worker shutdown
+                final List<CompletableFuture<Boolean>> periodWorkerShutdown = new ArrayList<>();
+                for (final List<ActorRef> workers : _periodWorkerActors.values()) {
+                    periodWorkerShutdown.addAll(_aggregator.shutdownActors(workers, PoisonPill.getInstance()));
+                }
+
+                // Wait for shutdown
+                CompletableFutures.allOf(periodWorkerShutdown).get(
+                        SHUTDOWN_TIMEOUT.toMillis(),
+                        TimeUnit.MILLISECONDS);
+
+                // Report any failures
+                final boolean success = periodWorkerShutdown.stream()
+                        .map(f -> f.getNow(false))
+                        .reduce(Boolean::logicalAnd)
+                        .orElse(true);
+                if (!success) {
+                    LOGGER.error()
+                            .setMessage("Failed stopping one or more period worker actors")
+                            .addData("aggregatorActor", this.self())
```

Comment:
nit. a few places in this file use `self()` and others use `this.self()`. Can we standardize on the former?

Reply:
Done!